src/legacy/core_plugins/kibana/public/visualize/np_ready/editor/editor.js

```diff
@@ -451,10 +460,18 @@ function VisualizeAppController(
       updateSavedQueryFromUrl(state.savedQuery);
 
       // if the browser history was changed manually we need to reflect changes in the editor
-      if (!_.isEqual(vis.getState(), state.vis)) {
-        vis.setState(state.vis);
-        vis.forceReload();
-        vis.emit('updateEditor');
+      if (
+        !_.isEqual(visualizations.convertFromSerializedVis(vis.serialize()).visState, state.vis)
+      ) {
+        vis.setState({
+          params: state.vis.params,
```

Comment:
`title` and `type` are missed here, they could be changed in the url string, but preserved in saved object as different values.

Reply:
i can't find title in the url ?

Reply:
nevermind, fixed